src/dmd/initsem.d

```diff
@@ -144,7 +144,13 @@ extern(C++) Initializer initializerSemantic(Initializer init, Scope* sc, Type t,
                             error(initLoc, "`%s` is not a member of `%s`", id.toChars(), sd.toChars());
                         return new ErrorInitializer();
                     }
-                    s = s.toAlias();
+                    auto s_alias = s.toAlias();
+                    if (s.isDeprecated())
+                        deprecation(i.loc, "field `%s` of struct `%s` is deprecated", s.toChars(), sd.toChars());
+                    else if (s_alias.isDeprecated())
+                        deprecation(i.loc, "field `%s` of struct `%s` is deprecated", s_alias.toChars(), sd.toChars());
```

Comment:
`s/field/alias/` ?

Also does this take into account deprecated scope ?

Reply:
> s/field/alias/ ?

No, think about library objects, you don't care if whether the symbol is an alias or not, you just know it is deprecated. However, I just noticed that if you have an alias to a deprecated symbol then you already get a deprecation at the alias declaration site, rendering this message redundant. I will delete the second check.

> Also does this take into account deprecated scope ? Doesn't seem to, it should probably call checkDeprecated.

Yes, it does.

Reply:
> Yes, it does.

Can you add a test then ?

Reply:
Done